### PR TITLE
CLDR-14386 Modernize List Users; work in progress

### PR DESCRIPTION
**Pull request by btangmu**
-Get more features working, including set locales command

-Make cldrListUsers a thin wrapper around cldrAccount

-Add unit tests

-Add script pretty.sh to run npx prettier on js files

-Comments, clean-up

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-14386
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

File: tools/cldr-apps/src/main/webapp/js/new/cldrListUsers.js [hunk 1]
```diff
@@ -8,994 +8,33 @@
  * and to hide everything else, minimizing global scope pollution.
  */
 const cldrListUsers = (function () {
-  const SHOW_GRAVATAR = false; // for debugging; TODO: restore to true for production
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/new/cldrListUsers.js:11]: I moved almost all of cldrListUsers.js into cldrAccount.js

File: tools/cldr-apps/src/main/webapp/js/new/cldrText.js [hunk 2]
```diff
@@ -286,7 +286,7 @@ const cldrText = (function () {
       "${count} old winning votes were automatically imported",
     "v-title_desc":
       "This area shows the date before which votes are considered “old”.",
-    special_account: "Settings",
+    special_account: "Account Settings",
     special_admin: "Admin Panel",
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/new/cldrText.js:290]: "Account Settings" is now the top-level description for both "My Account, Settings" and "List ... Users"

**Comment by btangmu** [comment]
To make calling `pretty.sh` more convenient, I have this in `~/.bash_profile`:
```
alias cdcldr='cd ~/Documents/WenlinDocs/Organizations/Unicode/CLDR_job/cldr'
alias cldrpretty='cdcldr && sh tools/cldr-apps/js-unittest/pretty.sh'
```

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserList.java [hunk 3]
```diff
@@ -332,18 +255,18 @@ else if (action.equals(LIST_ACTION_SHOW_PASSWORD)) {
                                 if (theirLocales == null) {
                                     theirLocales = "";
                                 }
-                                // TODO: LIST_ACTION_SETLOCALES -- compare above "LIST_ACTION_SETLOCALES + theirTag"???
-                                /***
-                                ctx.println("<label>Locales: (space separated) <input id='" + LIST_ACTION_SETLOCALES + theirTag + "' name='"
+                                String s = "<label>Locales: (space separated) <input id='"
+                                    + LIST_ACTION_SETLOCALES + theirTag + "' name='"
```
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserList.java:259]: Ideally this html should be generated on the front end...

File: tools/cldr-apps/src/main/webapp/js/new/cldrGear.js [hunk 4]
```diff
@@ -179,99 +179,100 @@ const cldrGear = (function () {
   let alreadySet = false;
 
   function set(gearMenuItems) {
-    if (!alreadySet) {
-      alreadySet = true;
```
Comment (btangmu) [inline tools/cldr-apps/src/main/webapp/js/new/cldrGear.js:183]: The only change in this file is to return early if alreadySet, and reduce the indentation for the rest of the function

File: tools/cldr-apps/js-unittest/pretty.sh [hunk 5]
```diff
@@ -0,0 +1,2 @@
+npx prettier --write tools/cldr-apps/src/main/webapp/js/new/*.js
+npx prettier --write tools/cldr-apps/js-unittest/*.js
```
Comment (srl295) [inline tools/cldr-apps/js-unittest/pretty.sh:2]: this is fine, but can also be done within npm with no .sh needed (could be called from maven)

```diff
diff --git a/tools/cldr-apps/js/package.json b/tools/cldr-apps/js/package.json
index 22c6369078..8cafacf354 100644
--- a/tools/cldr-apps/js/package.json
+++ b/tools/cldr-apps/js/package.json
@@ -6,7 +6,8 @@
   "scripts": {
     "test": "exit 0",
     "lint": "exit 0",
-    "build": "webpack"
+    "build": "webpack",
+    "prettier": "npx prettier --write ../src/main/webapp/js/new/*.js ../js-unittest/*.js"
   },
   "repository": {
     "type": "git",
```

File: tools/cldr-apps/src/main/webapp/js/new/cldrGear.js (hunk 4 above)
Comment (srl295) [reply tools/cldr-apps/src/main/webapp/js/new/cldrGear.js:183]: good, turned off whitespace for my review

**Review by srl295 — APPROVED** [review]
lgtm.

File: tools/cldr-apps/js-unittest/pretty.sh (hunk 5 above)
Comment (btangmu) [reply tools/cldr-apps/js-unittest/pretty.sh:2]: sounds good -- how would you then invoke it manually?

Comment (srl295) [reply tools/cldr-apps/js-unittest/pretty.sh:2]: @btangmu 

`npm run prettier`  in the `js` dir

**Pull request merged**